### PR TITLE
[sc-8062] Add ThoughtSpot extractor

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

To extract metadata from ThoughtSpot using [ThoughtSpot API](https://try-everywhere.thoughtspot.cloud/v2/#/everywhere/).

### 🤓 What?

- Add ThoughtSpot extractor

### 🧪 Tested?

Run extractor locally, and ingest MCE successfully in local environment.

### CODE REVIEW

**Review by mars-lan — COMMENTED** [review]
Also bump up version in pyproject?

File: metaphor/thought_spot/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# ThoughtSpot
+
+This connector extracts technical metadata from ThoughtSpot using [ThoughtSpot API](https://try-everywhere.thoughtspot.cloud/v2/#/everywhere)
+
+## Setup
+
+We recommend creating a secret key for the connector.
+
+1. Log into your ThoughtSpot instance.
```
Comment (mars-lan) [inline metaphor/thought_spot/README.md:9]: Does the user need to be an admin?

File: metaphor/thought_spot/config.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+@dataclass
+class ThoughtspotRunConfig(BaseConfig):
+    user: str
+
+    # ThoughtSpot instance url
+    base_url: str
+
+    password: Optional[str] = None
```
Comment (mars-lan) [inline metaphor/thought_spot/config.py:15]: Needs to add a validator to ensure that either `password` or `secret_key` is set (but not both).

File: metaphor/thought_spot/models.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
+from enum import Enum
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel
+
+
+class Header(BaseModel):
```
Comment (mars-lan) [inline metaphor/thought_spot/models.py:7]: Please link each model to the API doc if possible.

File: metaphor/thought_spot/utils.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+def mapping_data_object_type(type_: SourceType) -> ThoughtSpotDataObjectType:
+    mapping = {
+        SourceType.WORKSHEET: ThoughtSpotDataObjectType.WORKSHEET,
+        SourceType.ONE_TO_ONE_LOGICAL: ThoughtSpotDataObjectType.TABLE,
+        SourceType.AGGR_WORKSHEET: ThoughtSpotDataObjectType.VIEW,
+    }
+    return mapping.get(type_)
+
+
+def mapping_chart_type(type_: str) -> Optional[ChartType]:
```
Comment (mars-lan) [inline metaphor/thought_spot/utils.py:62]: Link to doc?

File: metaphor/thought_spot/config.py (hunk 2 above)
Comment (elic-eon) [reply metaphor/thought_spot/config.py]: Added a validator and tested it.

File: metaphor/thought_spot/models.py (hunk 3 above)
Comment (elic-eon) [reply metaphor/thought_spot/models.py:7]: This is still not possible. Metadata models were not provided by ThoughtSpot.

File: metaphor/thought_spot/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# ThoughtSpot
+
+This connector extracts technical metadata from ThoughtSpot using [ThoughtSpot API](https://try-everywhere.thoughtspot.cloud/v2/#/everywhere)
```
Comment (mars-lan) [inline metaphor/thought_spot/README.md:3]: Let's also mention that we're using V2 and points to https://try-everywhere.thoughtspot.cloud/v2/#/everywhere/api/rest/playgroundV2

File: metaphor/thought_spot/models.py (hunk 3 above)
Comment (mars-lan) [reply metaphor/thought_spot/models.py:7]: Got it. Seems like the sdk doesn't include any of these models. Could you please add a comment at the top for posterity? Thanks.

**Review by mars-lan — APPROVED** [review]
Great work!

**Pull request merged**